### PR TITLE
[FLINK-19993] Remove flink-connector-filesystem module

### PR DESCRIPTION
**Pull request by kl0u**
## What is the purpose of the change

The flink-connector-filesystem module contains (only) the deprecated BucketingSink. The BucketingSink. The sink is deprecated since FLINK 1.9 in favour of the relatively recently introduced StreamingFileSink.

For the sake of a clean and more manageable codebase, the community chose to remove it [1] after the discussion in [2].

[1]
https://lists.apache.org/thread.html/red2bc04c5d60a6a923ea499a49d7889d66661c31ac1987ea6d9a3fe5%40%3Cdev.flink.apache.org%3E
[2] https://lists.apache.org/thread.html/re24ceedc02402ac9a6ce1e07b690852320a265b081f416ebac543aaf%40%3Cuser.flink.apache.org%3E

## Brief change log

Removed the module itself and wherever it appeared in the documentation.

## Verifying this change

This change is a code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (**yes** / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by kl0u** [comment]
Still waiting for AZP to give the green light.

**Comment by kl0u** [comment]
@flinkbot run azure

File: flink-end-to-end-tests/run-nightly-tests.sh [hunk 1]
```diff
@@ -184,9 +184,7 @@ run_test "Batch SQL end-to-end test" "$END_TO_END_DIR/test-scripts/test_batch_sq
 run_test "Streaming SQL end-to-end test (Old planner)" "$END_TO_END_DIR/test-scripts/test_streaming_sql.sh old" "skip_check_exceptions"
 run_test "Streaming SQL end-to-end test (Blink planner)" "$END_TO_END_DIR/test-scripts/test_streaming_sql.sh blink" "skip_check_exceptions"
 # skip test if hadoop version is 2.4.1 (FLINK-16629)
```
Comment (dawidwys) [inline flink-end-to-end-tests/run-nightly-tests.sh:186]: remove the comment as well

**Review by dawidwys — APPROVED** [review]
+1 for the PR once azure gives green. Had just one minor comment.

**Comment by kl0u** [comment]
Thanks for the review @dawidwys ! Merged

**Pull request closed without merging**